testapp/tests.py

```diff
@@ -60,86 +62,187 @@ def test_create_change_request(self, mock_get_snow_group_guid, mock_pysnow):
             'assignment_group': {'value': uuid.uuid4()},
             'state': '2'
         }
-        self.mock_pysnow_client.insert.return_value = fake_insert_retval
+
+        fake_resource = mock.MagicMock()
+        fake_resource.create.return_value = fake_insert_retval
+
+        self.mock_pysnow_client.resource.return_value = fake_resource
         mock_pysnow.Client.return_value = self.mock_pysnow_client
```

Comment:
Can you get rid of the `self.mock_pysnow_client`? Since it's just an empty mock, it's not extremely useful, and it's making this logic a bit hard to follow.

Reply:
The `self.mock_pysnow_client` is being used in almost all the test cases. So instead of instantiating it every time within the test cases themselves, I think its better to have it at the setUp() method.